test/regression.uts

```diff
@@ -6398,6 +6398,43 @@ pkts[0].show()
 assert HTTPResponse in pkts[0]
 assert pkts[0].load == b'This is a test file for testing brotli decompression in Wireshark\n'
 
+= HTTP decompression (zstd)
+~ zstd
+
+conf.debug_dissector = True
+load_layer("http")
+
+import os
+import zstandard
+
+# sample server: $ socat -v TCP-LISTEN:8080,fork,reuseaddr SYSTEM:'(echo -ne "HTTP/1.1 200 OK\r\nContent-Encoding: zstd\r\n\r\n") > tmp && dd bs=1G count=1 status=none | zstd --stdout >> tmp && cat tmp'
+# sample client: $ curl -v localhost:8080/tmp_echo_zstd_request_for_testing -o a.html
+tmp = "/test/pcaps/http_compressed-zstd.pcap"
+filename = os.path.abspath(os.path.join(os.path.dirname(__file__),"../")) + tmp
+filename = os.getenv("SCAPY_ROOT_DIR")+tmp if not os.path.exists(filename) else filename
+
+# First without auto decompression
+
+conf.contribs["http"]["auto_compression"] = False
+pkts = sniff(offline=filename, session=TCPSession)
+
+data = b'\xfd\x2f\xb5\x28\x34\x55\x80\x41\x90\x6f\x20\x01\xb8\x97\x01\xc1\xa2\x14\xff\x69\xad\xa2\x8b\xb6\x8b\x83\x41\xcd\x78\x2d\x21\xfe\x53\x68\xd0\x13\x80\xf1\xd5\x78\x6a\x9a\x91\x0d\x43\xd9\x63\x22\xba\x92\xdc\xea\x89\x49\x20\x1e\xcb\x72\x31\x72\x39\x66\xe9\xf9\x75\x21\x3b\x6e\xf6\x14\xb2\xf0\x70\x57\x9d\x4c\x87\x7b\x2d\x35\x10\x92\x2a\xe6\xd9\x8d\x68\x73\xf7\xbc\x41\xef\x7e\x07\x84\xd9\x1d\x14\x88\xe8\x60\xbe\xfa\xf1\x28\x66\xb1\x04\x05\xb6\xd2\x52\x29\x00\xfa'
+
+pkts[0].show()
+assert HTTPResponse in pkts[0]
+assert pkts[0].Content_Encoding == b'zstd'
+assert pkts[0].Content_Type == b'text/plain'
+assert pkts[0].load == data
+
+# Now with auto decompression
+
+conf.contribs["http"]["auto_compression"] = True
+pkts = sniff(offline=filename, session=TCPSession)
```

Comment:
If you're using `TCPSession`, the packets are only returned when complete. Your pcap only contains a single TCP fragment and not the whole HTTP frame, so you likely just want to remove the use of `TCPSession`.